scheduler/src/cook/mesos/util.clj

```diff
@@ -198,6 +198,39 @@
      get-pending-jobs-duration
      (get-pending-job-ents* unfiltered-db true))))
 
+(defn generate-intervals
+  "Generates a list of intervals between start and end as pairs
+   [interval-start interval-end].
+
+   Parameters:
+   ----------
+   start : clj-time/datetime
+   end : clj-time/datetime
+   interval->steps : fn
+      function that takes a clj-time/interval and returns the number of steps
+      based on the step-fn between start and end
+   step-fn : fn
+      function that returns a clj-time/timedelta given n steps
+
+   Returns:
+   --------
+   list of pairs, [interval-start interval-end]"
+  ([start end]
+   (generate-intervals start end t/in-days t/days))
+  ([start end interval->steps step-fn]
```

Comment:
Do you need `interval->steps`? Can you just iterate `step-fn` until you walk past `end`?